src/weapon.py

```diff
@@ -15,12 +14,13 @@ class Weapon:
     car: arcade.Sprite
     time_since_shoot: float
     weapon_icon: arcade.texture
+    muzzle_offset: Tuple[float, float, float]
```

Comment:
I think there's probably a better name/class for this.  Probably needs to be it's own ticket since this is for interfacing with shared code

Reply:
The academically better term is `Transform` or `Transformation`, which is also the idiomatic terminology used in game engines.  This is a geometric transformation applied to the parent's position to derive the muzzle's position.

Since objects in games are usually organized in a hierarchy, only the root object's transformation represents an absolute world position.  Every nested object's transformation is applied relative to the parent's position.

Reply:
`weapon_to_muzzle_transform`?